pyartcd/pyartcd/jenkins.py

```diff
@@ -1,71 +1,107 @@
-import base64
 import logging
 import os
-import aiohttp
+from enum import Enum
+
+from jenkinsapi.jenkins import Jenkins
+
+from pyartcd import constants
 
 logger = logging.getLogger(__name__)
 
 
-async def trigger_jenkins_job(job_path: str, params: dict = {}):
+class Jobs(Enum):
+    BUILD_SYNC = 'aos-cd-builds/build%2Fbuild-sync'
+    MICROSHIFT = 'aos-cd-builds/build%2Fbuild-microshift'
+    OCP4 = 'aos-cd-builds/build%2Focp4'
+    RHCOS = 'aos-cd-builds/build%2Frhcos'
+
+
+logger.info('Initializing Jenkins client')
+jenkins = Jenkins(
+    constants.JENKINS_SERVER_URL,
+    username=os.environ['JENKINS_SERVICE_ACCOUNT'],
+    password=os.environ['JENKINS_SERVICE_ACCOUNT_TOKEN'],
+    lazy=True)
+logger.info('Connected to Jenkins %s', jenkins.version)
+
+
+def allow_backlinking(params: dict) -> dict:
     """
-    Trigger a job using remote API calls.
-    :param job_path: relative path to the job, starting from <buildvm hostname>:<jenkins port>
-    :param params: optional dict containing job parameters
+    If 'BUILD_URL' env var was provided, add a TRIGGERED_FROM
+    field to the build params to allow backlinking to the original job
     """
 
-    service_account = os.environ['JENKINS_SERVICE_ACCOUNT']
-    token = os.environ['JENKINS_SERVICE_ACCOUNT_TOKEN']
-
     build_url = os.environ.get('BUILD_URL', '')
```

Comment:
I guess we are exploring other ways to automatically get the parent build..
I would enforce this, so complain if build_url isn't set - since I can't
think of a <not from a jenkins job invokation> of this func

Reply:
Done in https://github.com/openshift-eng/aos-cd-jobs/pull/3663/commits/91e809f1c6701514533552a1fcdb8115bcfc1df8